openapi/src/main/java/io/micronaut/openapi/view/OpenApiViewConfig.java

```diff
@@ -309,10 +310,31 @@ public void setTitle(String title) {
     /**
      * Returns the relative openApi specification url path.
      *
+     * @param context Visitor context.
+     *
      * @return A path.
      */
-    public String getSpecURL() {
-        return StringUtils.prependUri(serverContextPath, StringUtils.prependUri(mappingPath, specFile));
+    public String getSpecURL(VisitorContext context) {
+        String specUrl = StringUtils.prependUri(serverContextPath, StringUtils.prependUri(mappingPath, specFile));
+        if (StringUtils.isEmpty(serverContextPath)) {
+            String contextPath = OpenApiApplicationVisitor.getConfigurationProperty("micronaut.server.context-path", context);
```

Comment:
Hmm I just realized that this probably breaks incremental compilation with this module

Reply:
@graemerocher hmm... Then I have a few questions:

Why did you come to this conclusion?
How can we check this?
And how bad is it to make this annotation processor non-incremental?

Reply:
well this value should generally be supplied as an annotation processor option so that when it changes it triggers a recompilation, but here it is read from a configuration file?

Perhaps it is not an issue and generally this module is not well designed for incremental compilation.

Reply:
@graemerocher In my implementation, the compiler does not run. The fact of the matter is that now it's just reading the yml file with the settings. So far, we do not need access to Trees, i.e. in my opinion, we now have an isolating level, not dynamic . But I'm not familiar with incremental compilation, so I can't guarantee anything :-)